### PR TITLE
Make CultureDictionary IEnumerable

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: test/OrchardCore.Tests/Localization/CultureDictionaryTests.cs [hunk 1]
```diff
@@ -50,5 +53,28 @@ public void IntexerThrowsPluralFormNotFoundExceptionIfSpecifiedPluralFormDoesntE
...
+
+        [Fact]
+        public void EnumerateCultureDictionary()
+        {
+            // Arrange
+            var dictionary = new CultureDictionary("ar", _arPluralRule);
+            dictionary.MergeTranslations(new List<CultureDictionaryRecord>
+            {
+                new CultureDictionaryRecord("Hello", null, new[] { "مرحبا" }),
+                new CultureDictionaryRecord("Bye", null, new[] { "مع السلامة" })
```
Comment (sebastienros) [inline test/OrchardCore.Tests/Localization/CultureDictionaryTests.cs:65]:
```suggestion
                new CultureDictionaryRecord("Hello", "مرحبا"),
                new CultureDictionaryRecord("Bye", "مع السلامة")
```

**Comment by hishamco** [comment]
@sebastienros is it fine for you to merge this?

**Comment by agriffard** [comment]
@jtkech Can you please review?

**Comment by jtkech** [comment]
Okay will do soon ;)

**Comment by sebastienros** [comment]
Not sure why you need it but it doesn't hurt

**Review by sebastienros — APPROVED** [review]

**Comment by jtkech** [comment]
As i remember @hishamco was talking about perf but not sure ;)

**Comment by hishamco** [comment]
Not about Perf @jtkech ;) that's another issue, the idea to make `CultureDictionary` foreachable which is useful in some cases

**Comment by jtkech** [comment]
Okay cool then

Yes because i didn't see a current usage in the code

**Comment by hishamco** [comment]
Perhaps there's no use case I'm the source at that time but I need to check now. If there's not at least you can iterate over this collection like the other once

**Comment by sebastienros** [comment]
> Perhaps there's no use case

I asked because this library can be used outside of OC, so I assumed there was some use case. But if you don't even have one then don't do anything. The only thing you can do is introduce bugs, or create more public surface to maintain.

**Comment by hishamco** [comment]
Seem I made a typo in my last reply, what I mean is perhaps there's no use case in the source code at that time but there are some use cases like:

- Po Parser could return a `CultureDictionary`

- Filter the translations in `CultureDictionary` , this could be useful if I want to provide a UI for translation per component or view

there could be more use cases outside OC

**Comment by sebastienros** [comment]
Ok, and this change doesn't seem risky. But it was good to talk about when to not do anything.

**Comment by hishamco** [comment]
You are right .. at the time of writing the PR (7 months ago) there's a case where I need to traverse the `CultureDictionary` records I find there's no proper way to do that ;)

I think making `PoParser` returns `CultureDictionary` will be a proof / use case for this PR if you agreed with

Thanks

**Pull request merged**